Exercise/compute_lowest_terms.py

```diff
@@ -1,4 +1,122 @@
 ## TODO: complete the function "lowest_terms" below
 
 def lowest_terms(x):
-	return ""
+	#convert the given string into two integers
+	num, denom = string_to_nums(x)
+
+	if num == 0:
+		return '0'
+	elif denom == 0:
+		return 'Undefined'
+	else:
+		#find the lowest terms for num and denom
+		num, denom = find_lowest_terms(num, denom)
+		#convert terms from integers to string
+		result = nums_to_string(num, denom)
+		return result
+
+
+def string_to_nums(string):
+	"""Convert given fraction from string into integers
+
+	INPUT:
+	string - Fraction in string form
+
+	OUTPUT:
+	splitted_nums - A list consisting of two integers representing the numerator and denominator of given fraction respectively
+	"""
+
+	#split string
+	splitted_strings = string.split('/')
+	#convert splitted strings to integers
+	splitted_nums = [int(string) for string in splitted_strings]
+
+	return splitted_nums
+
+
+def nums_to_string(num, denom):
+	"""Convert given integers into a fraction in string form
+
+	INPUT:
+	num - The numerator of the fraction
+	denom - Denominator of the fraction
+
+	OUTPUT:
+	string - fraction in string form
+	"""
+
+	string = str(num) +'/'+ str(denom)
+	return string
+
+
+def highest_possible_common_factor(num, denom):
+	""" Find the highest possible common factor between two numbers
+
+	The highest possible common factor of given set of numbers is a number that might or might not be
+	the Highest Common Factor of those numbers, but above which the given numbers has no common factor
+
+	INPUT:
+	num - The first number
+	denom - The second number
+
+	OUTPUT:
+	hpcf - The highest possible common factor of the two given numbers
+	"""
+
+	#find the smallest between num and denom
+	smallest = num if abs(num) < abs(denom) else denom
```

Comment:
good! note however that built-in functions are almost always better optimized (run faster). 

**using built-in-function min()**
smallest = min(abs(num), abs(denom))